bulbs/infographics/validators.py

```diff
@@ -0,0 +1,47 @@
+import re
+import six
+
+from django.core.exceptions import ValidationError
+from django.utils.deconstruct import deconstructible
+from django.utils.encoding import force_text
+from django.utils.functional import SimpleLazyObject
+
+
+def _lazy_re_compile(regex, flags=0):
+    """Lazily compile a regex with flags."""
+    def _compile():
+        # Compile the regex if it was not passed pre-compiled.
+        if isinstance(regex, six.string_types):
+            return re.compile(regex, flags)
+        else:
+            assert not flags, "flags must be empty if regex is passed pre-compiled"
+            return regex
+    return SimpleLazyObject(_compile)
+
+
+@deconstructible
+class ColorValidator(object):
+    message = ('Enter a valid hexadecimal color.')
+    code = 'invalid'
+    color_regex = _lazy_re_compile(
```

Comment:
uncompiled regexes are still pretty fast in python, not sure if it's worth the extra code to pre-compile -- your call.